ocp-all-inclusive/main.tf

```diff
@@ -0,0 +1,117 @@
+##############################################################################
+# base-ocp-vpc-module
+##############################################################################
+
+locals {
+  # Input variable validation
+  # tflint-ignore: terraform_unused_declarations
+  validate_cos_inputs = (var.use_existing_cos == false && var.cos_name == null) ? tobool("A value must be passed for var.cos_name if var.use_existing_cos is false.") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_existing_cos_inputs = (var.use_existing_cos == true && var.existing_cos_id == null) ? tobool("A value must be passed for var.existing_cos_id if var.use_existing_cos is true.") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_kp_inputs = (var.existing_key_protect_instance_guid == null && var.existing_key_protect_root_key_id != null) || (var.existing_key_protect_root_key_id != null && var.existing_key_protect_instance_guid == null) ? tobool("To enable encryption, values must be passed for both var.existing_key_protect_instance_guid and var.existing_key_protect_root_key_id. Set them both to null to create cluster without encryption (not recommended).") : true
+
+  # If encryption enabled generate kms config to be passed to cluster
+  kms_config = var.existing_key_protect_instance_guid != null && var.existing_key_protect_root_key_id != null ? {
+    crk_id           = var.existing_key_protect_root_key_id
+    instance_id      = var.existing_key_protect_instance_guid
+    private_endpoint = var.key_protect_use_private_endpoint
+  } : null
+}
+
+module "ocp_base" {
+  source                          = "git::https://github.ibm.com/GoldenEye/base-ocp-vpc-module.git?ref=4.0.1"
```

Comment:
base-ocp-vpc-module should be pushed out to terraform-ibm-module and referenced from there